1-common-tasks/algorithms/accumulate-values.cpp

```diff
@@ -0,0 +1,49 @@
+// Accumulate values from a range
+// C++11
+
+#include <deque>
+#include <forward_list>
+#include <numeric>
+#include <vector>
+
+template <typename Collection>
+typename Collection::value_type
+sum_numbers(Collection const& collection)
+{
+    using ReturnType = typename Collection::value_type;
+    return std::accumulate(std::begin(collection), std::end(collection), static_cast<ReturnType>(0));
```

Comment:
Well, it's a generic free function that will call what you expect without any overload (for any decent compiler). It also support arrays, so you can call sum_numbers on an array as well. And of course you can call it with any non-STL container for which an appropriate overload of the functions has been created.